registry/handlers/referrers.go

```diff
@@ -190,3 +177,57 @@ func readlink(ctx context.Context, path string, stDriver driver.StorageDriver) (
 
 	return digest.Parse(string(content))
 }
+
+func getArtifactReferrers(ctx context.Context,
+	blobStatter distribution.BlobStatter,
+	referrerDigest digest.Digest,
+	man distribution.Manifest,
+	referrers *[]v1.Descriptor,
+	artifactType string) error {
+	artifactManifest, ok := man.(*ociartifact.DeserializedManifest)
+	if !ok {
+		return fmt.Errorf("the given manifest is not of type ociartifact.DeserializedManifest")
+	}
+	extractedArtifactType := artifactManifest.ArtifactType
+	// filtering by artifact type or bypass if no artifact type specified
+	if artifactType == "" || extractedArtifactType == artifactType {
+		desc, err := blobStatter.Stat(ctx, referrerDigest)
+		if err != nil {
+			return err
+		}
+		desc.MediaType, _, _ = man.Payload()
+		artifactDesc := v1.Descriptor{
+			MediaType:    desc.MediaType,
+			Size:         desc.Size,
+			Digest:       desc.Digest,
+			ArtifactType: extractedArtifactType,
+			Annotations:  artifactManifest.Annotations,
+		}
+		*referrers = append(*referrers, artifactDesc)
+	}
+	return nil
+}
+
+func getImageReferrers(ctx context.Context,
```

Comment:
In Image Manifest, the media type of the config is used as the artifact type.

Reply:
Added filtering with config media type.